src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -897,4 +904,30 @@ public void setSpectatorTarget(@Nullable org.spongepowered.api.entity.Entity ent
         return event;
     }
 
+    @Override
+    public Optional<WorldBorder> getWorldBorder() {
+        return Optional.ofNullable(this.worldBorder);
+    }
+
+    @Override
+    public void setWorldBorder(@Nullable WorldBorder border, Cause cause) {
+        if (this.worldBorder == border) {
+            return; //do not fire an event since nothing would have changed
+        }
+        if (!SpongeImpl.postEvent(SpongeEventFactory.createChangeWorldBorderEventTargetPlayer(cause, Optional.ofNullable(this.worldBorder), Optional.ofNullable(border), this))) {
+            if (border == null) { //is the list about to be unset?
+                ((net.minecraft.world.border.WorldBorder) this.worldBorder).listeners.remove(this.borderListener); //remove the listener, if so
+            }
+            this.worldBorder = border;
+            if (this.worldBorder != null) {
+                if (!((net.minecraft.world.border.WorldBorder) this.worldBorder).listeners.contains(this.borderListener)) {
```

Comment:
The check is necessary as the `border` variable represents a different `WorldBorder` object (if it were the same object, the listener would already be set and the method would have already exited by this point) so the listener may not be added to the new border. This check makes it so that a border listener is always present on the world border.